cvmfs/server/cvmfs_server_util.sh

```diff
@@ -733,6 +733,54 @@ has_selinux() {
   $GETENFORCE_BIN | grep -qi "enforc" || return 1
 }
 
+# Functions to retrieve information about the operating system.
+# Used in cvmfs_server_json.sh to populate meta.json.
+# Fails gracefully if no os-release file is found (values become empty strings)
+_os_set_etc_release_filename() {
+	_CVMFS_OS_RELEASE_FILENAME=''
+	if test -e "/etc/os-release"; then
+		_CVMFS_OS_RELEASE_FILENAME="/etc/os-release";
+	elif test -e "/usr/lib/os-release"; then
+		_CVMFS_OS_RELEASE_FILENAME="/usr/lib/os-release";
+	fi
+}
+
+# Pick a specific field from the os-release file and return it. 
+# 1. _os_set_etc_release_filename is called on every check, this is probably overkill,
+#    but the question would be where to call this once and for all. In theory one could
+#    call it right after the declaration above?
+# 2. If a fieldname isn't provided, die due to wrongful usage of the (internal) function.
+# 3. If we couldn't find an os-release file, return nothing.
+# 4. If we grep fails to find anything, return nothing. We may want to check against known
+#    keys from https://www.freedesktop.org/software/systemd/man/os-release.html?
+_os_etc_release_get_field() {
+	_os_set_etc_release_filename
+
+	fieldname=$1
+	if [ "x${fieldname}" == "x" ]; then
+		die "Internal error: _os_etc_release_get_field expects a field name to search for!"
+	fi
+
+	if [ "x${_CVMFS_OS_RELEASE_FILENAME}" == "x" ]; then
+		# If we are unable to find a proper /etc/os-release file return nothing.
+		:
+	else
+		grep "^${fieldname}=" ${_CVMFS_OS_RELEASE_FILENAME} | cut -f2 -d'"'
```

Comment:
Suggestion: `eval $(sed -n "/^${fieldname}=/p" ${_CVMFS_OS_RELEASE_FILENAME})`

Reply:
Huh. Interesting approach. Saves a pipe but introduces an eval. Is there a specific reason for wanting this approach? I don't mind either way, I'm mostly asking to learn something. :)

Reply:
The grep..cut isn't going to work if the double-quotes are missing.  I see that is the case on debian for ID, where it is `ID=debian`.

On the other hand, the eval does not print to stdout as expected.  This one works for me:
```
eval echo $(sed -n "s/^${fieldname}=//p" /etc/os-release)
```

Reply:
Oh I thought that looked familiar.  I wrote that same line [before](https://github.com/cvmfs-contrib/config-repo/blob/osg/libexec/osgstorage-auth.conf#L20).

Reply:
Aha. :) https://github.com/cvmfs/cvmfs/pull/2863/commits/703dad1b9603d5fca9ee3d6359e3f5496f79c0ca now uses the line from @DrDaveD. Do note that supposedly the os-release-file may be located under `/usr/lib`.